packages/cli/src/index.ts

```diff
@@ -0,0 +1,748 @@
+import Path from "path";
+import fs from "fs";
+
+import chalk from "chalk";
+
+import { convertType } from "ts-to-zod";
+import {
+  ConvertTypeContext,
+  SchemaGenContext,
+  ImportInfo,
+  convertSymbol,
+  ErrorAbort,
+  Incident,
+  Diagnostics,
+} from "ts-to-zod/dist/convertType";
+
+import {
+  generateFiles,
+  generateImportStatementsESM,
+  generatePath,
+} from "ts-to-zod/dist/generateFiles";
+
+import {
+  GenOptions,
+  createGenerationConfig,
+} from "ts-to-zod/dist/filePathConfig";
+
+import { Watcher } from "./watch";
+import {
+  EndpointTypeInstance,
+  NodeType,
+  preprocessEndpoint,
+} from "./endpointPreprocess";
+
+import { program as argParser } from "commander";
+import {
+  convertPathToImport,
+  isSymbolStlMethod,
+  mangleString,
+  statOrExit,
+} from "./utils";
+
+// TODO: add dry run functionality?
+argParser.option("-w, --watch", "enables watch mode");
+argParser
+  .option(
+    "-d, --directory <path>",
+    "the directory to generate schemas for",
+    "."
+  )
+  .allowExcessArguments(false);
+
+const NODE_MODULES_GEN_PATH = "@stl-api/gen";
+
+interface CallDiagnostics {
+  line: number;
+  column: number;
+  filePath: string;
+  diagnostics: Map<string, Diagnostics>;
+}
+
+interface EndpointCall {
+  endpointPath: string;
+  schemaExpression: ts.Expression;
+}
+
+async function main() {
+  argParser.parse();
+  const options = argParser.opts();
+
+  const directoryStat = await statOrExit(options.directory);
+  if (!directoryStat.isDirectory()) {
+    console.error(`Error: '${options.directory}; is not a directory.`);
+    process.exit(1);
+  }
+
+  const packageJsonPath = await pkgUp({
+    cwd: options.directory,
+  });
+  if (!packageJsonPath) {
+    console.error(
+      `Folder ${Path.relative(
+        ".",
+        options.directory
+      )} and its parent directories do not contain a package.json.`
+    );
+    process.exit(1);
+  }
+  const rootPath = Path.dirname(packageJsonPath);
+
+  const tsConfigFilePath = Path.relative(
+    ".",
+    Path.join(rootPath, "tsconfig.json")
+  );
+  const tsConfigStat = await statOrExit(tsConfigFilePath);
+  if (!tsConfigStat.isFile()) {
+    console.error(`Error: '${tsConfigFilePath}' is not a file.`);
+  }
+
+  let watcher: Watcher | undefined;
+  if (options.watch) {
+    watcher = new Watcher(rootPath);
+  }
+
+  const project =
+    watcher?.project ||
+    new tm.Project({
+      // TODO: should file path be configurable?
+      tsConfigFilePath,
+    });
+
+  const baseCtx = new SchemaGenContext(project);
+
+  const generationOptions = {
+    genLocation: {
+      type: "node_modules",
+      genPath: NODE_MODULES_GEN_PATH,
+    },
+    rootPath,
+    zPackage: "stainless",
+  } as const;
+
+  const printer = tm.ts.createPrinter();
+
+  const succeeded = await evaluate(
+    project,
+    baseCtx,
+    generationOptions,
+    rootPath,
+    printer
+  );
+
+  if (watcher) {
+    for await (const { path } of watcher.getEvents()) {
```

Comment:
Hmm, not sure I understand the solution. We could also partially address the issue by setting a larger chokidar interval so that more changes are aggregated (right now it's 100ms, 1000ms seems reasonable)

Reply:
> Polling-specific settings (effective when usePolling: true)
interval (default: 100). Interval of file system polling, in milliseconds

Polling isn't the default, it uses filesystem events from the fs if possible.  And I don't see any evidence that changes are aggregated into one single event (you'd still be getting a separate event for each file that changed)